rstream/client.py

```diff
@@ -206,9 +227,12 @@ async def _listener(self) -> None:
 
             for _, handler in self._handlers.get(frame.__class__, {}).items():
                 try:
-                    maybe_coro = handler(frame)
-                    if maybe_coro is not None:
-                        await maybe_coro
+                    if frame.__class__ == schema.Deliver:
+                        await self._frames.put(FrameEntry(handler, frame))
```

Comment:
We can get a corresponding handler for a frame when receiving it from `self_frames` queue, it seems a bit redundant to put `handler` along with each frame

Reply:
HI @qweeze do you mean to get it from self._handlers in _run_delivery_handlers?

Reply:
Oh never mind I think I've understood now!

Reply:
> do you mean to get it from self._handlers in _run_delivery_handlers?

Yes, exactly